client/my-sites/email/email-management/home/utils.js

```diff
@@ -125,6 +113,11 @@ export function resolveEmailPlanStatus( domain, emailAccount, isLoadingEmails )
 	};
 
 	if ( hasGSuiteWithUs( domain ) ) {
+		// Check for pending TOS acceptance warnings at the account level
+		if ( hasGoogleAccountTOSWarning( emailAccount ) ) {
```

Comment:
Non-blocking, but #53092 also adds a similar check based on the `domain` object (which we're guaranteed to have) rather than `emailAccount`, which we may or may not have available to us in this function.

It may also be clearer to have this be `if ( emailAccount && hasGoogleAccountTOSWarning( emailAccount ) )` to reflect the dependency on `emailAccount`.